hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/ToOneAttributeMapping.java

```diff
@@ -998,6 +998,22 @@ public EntityFetch generateFetch(
 				|| fetchParent.getNavigablePath() instanceof TreatedNavigablePath
 				&& parentNavigablePath.equals( fetchParent.getNavigablePath().getRealParent() );
 
+		if ( cardinality == Cardinality.ONE_TO_ONE && bidirectionalAttributeName != null && fetchParent.getReferencedMappingType() == getEntityMappingType() ) {
+			return new EntityFetchSelectImpl(
+					fetchParent,
+					this,
+					fetchablePath,
+					foreignKeyDescriptor.createDomainResult(
+							fetchablePath,
+							parentTableGroup,
+							ForeignKeyDescriptor.Nature.TARGET,
+							creationState
+					),
+					true,
```

Comment:
Is this really always a select by unique key?

Reply:
it seems so for the case case of ` cardinality == Cardinality.ONE_TO_ONE && bidirectionalAttributeName != null && fetchParent.getReferencedMappingType() == getEntityMappingType()`